Uplift/Controllers/SportsDetailViewController.swift

```diff
@@ -9,23 +9,181 @@
+            cell.configure(for: post, dropStatus: dropStatus)
+            return cell
+        case .discussion:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: discussionSectionReuseIdentifier, for: indexPath) as! SportsDetailDiscussionCollectionViewCell
+            cell.configure(for: post)
+            return cell
+        case .comment(let c):
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: commentSectionReuseIdentifier, for: indexPath) as! SportsDetailCommentCollectionViewCell
+            cell.configure(for: c)
+            return cell
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, referenceSizeForHeaderInSection section: Int) -> CGSize {
+        return CGSize(width: collectionView.frame.width, height: 264)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        return 0
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let itemType = section.items[indexPath.item]
+        let width = collectionView.frame.width
+
+        switch itemType {
+        case .info:
+            return CGSize(width: width, height: 157)
+        case .players:
+            // TODO: add player names field to post.
+            let players = [User(id: "p1", name: "Zain Khoja", netId: "znksomething"),
+            User(id: "p2", name: "Amanda He", netId: "noidea"),
+            User(id: "p3", name: "Yi Hsin Wei", netId: "y???")]
+            let nameHeight = 24
+            let baseHeight = 82
+            let height = CGFloat(baseHeight + nameHeight * (dropStatus == .closed ? 1 : players.count))
+            return CGSize(width: width, height: height)
+        case .discussion:
+            return CGSize(width: width, height: 64)
+        case .comment(let c):
+            // https://stackoverflow.com/questions/30450434/figure-out-size-of-uilabel-based-on-string-in-swift
+            let horizontalPadding = 12
+            let imageSize = 32
+            let labelHeight = 16
+            let leadingPadding = 14
+            let textHorizontalPadding = 16
+            let textVerticalPadding = 8
+            let trailingPadding = 28
+            let verticalPadding = 4
+            let textWidth = width - CGFloat(leadingPadding + imageSize + horizontalPadding + textHorizontalPadding * 2 + trailingPadding)
+            let constraintRect = CGSize(width: textWidth, height: .greatestFiniteMagnitude)
+            let boundingBox = c.text.boundingRect(with: constraintRect, options: .usesLineFragmentOrigin, attributes: [.font: UIFont._12MontserratLight as Any], context: nil)
```

Comment:
i think we already have this function inside `String+Extensions.swift`: https://github.com/cuappdev/uplift-ios/blob/master/Uplift/Extensions/String%2BShared.swift